lib/chef/knife/ssh.rb

```diff
@@ -247,18 +247,18 @@ def search_nodes
       # @since 12.5.0
       # @param host [String] Hostname for this session.
       # @param port [String] SSH port for this session.
-      # @param type [Integer] Flag: host or gateway key
       # @param user [String] Optional username for this session.
+      # @param gatway [Boolean] Flag: host or gateway key
       # @return [Hash<Symbol, Object>]
-      def session_options(host, port, type = 0, user = nil)
+      def session_options(host, port, user = nil, gateway = false)
```

Comment:
`gateway: false`, rather than `gateway = false` - in the code above you're using a named argument, but here you're using a positional argument. In general, we prefer a named argument so if you can make the two line up I think we're good to go.